### PR TITLE
Pretraining

### PR DESCRIPTION
**Pull request by cssprad1**
# Pytorch-caney - Adding Satvision/SimMIM/SWIN pretraining

## Description
This PR adds the code base to enable pre-training of SWIN vision transformers using the Masked Image Modeling (MiM) of self-supervised contrastive learning. The implementation is broken up into several sub-modules. I tried best to match the style of of PTC is set up. 

## Testing
Needs independent testing

### CODE REVIEW

**Review by jordancaraballo — APPROVED** [review]
The new organization looks good. I will leave the comments for the next shorter review.
